### DISCUSSION TITLE
Add support for expiring `download_urls` and `download endpoint`

### DISCUSSION THREAD

**Discussion by davidwessman** [shrinerb/shrine#707] [Q&A]
Hello!

We would like to find some way to make expiring urls when using `download_url`, and check this with some signed parameter in the `Uploader.download_response`.

As far as I can see that is not possible right now, is there anyway to do it with the current implementation or any tips on how to do it? Would it make sense to add it to the gem?

**Comment by janko** [comment]
Yes, I would welcome a PR that adds this to the `download_endpoint` plugin. There is an implementation of expiring URLs in the `derivation_endpoint` plugin, maybe a similar strategy can be used, though that would require signing.

**Reply by davidwessman** [reply]
I wrote a PR that uses `ActiveSupport::MessageVerifier`, would love some feedback 🙂 
https://github.com/shrinerb/shrine/pull/708

**Comment by Faq** [comment]
Hello, I see the MR is ready and still not merged. I need similar functionality, where after few years download link should expire and file deleted.
Just wondering how you manage file here.
